### PR TITLE
feat: support trigger events option

### PR DESCRIPTION
**Pull request by Juiced66**
This PR is intended to support triggerEvents request parameter for embeded sdk. 

It is unit tested and will be functionnal tested when released on kuzzle side. 

A lot of doc as been realigned with code

JIRA: [KZLPRD-579](https://kuzzle.atlassian.net/browse/KZLPRD-579) 

[KZLPRD-579]: https://kuzzle.atlassian.net/browse/KZLPRD-579?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by etrousset — APPROVED** [review]

**Review by rolljee — COMMENTED** [review]
Everything look good and updated.

There is just something odd about the :any you added to the request.

Since you have the type for the KuzzleRequest isn't that possible to put the type instead of any ?

**Comment by Juiced66** [comment]
> Why didn't you put this (duplicated) piece of code in the `query` method directly?
> 
> ```js
> if (options.triggerEvents !== undefined) {
>   request.triggerEvents = options.triggerEvents;
> }
> ```
> 
> I see that stuff like this is already done in this method:
> 
> https://github.com/kuzzleio/sdk-javascript/blob/83e434b679f884604e567976967f31562ec5ba32/src/Kuzzle.ts#L901-L906

I've thinked about that, it was an option but not all methods got options and it could lead to useless code for those. I thought it was less readable to apply it were needed but more optimized. If we decide to apply it directly on query it is doable for sure

**Comment by Juiced66** [comment]
> Everything look good and updated.
> 
> There is just something odd about the :any you added to the request. Since you have the type for the KuzzleRequest isn't that possible to put the type instead of any ?
> 
> Or like Alex said, it could be possible to avoid duplication at the Kuzzle.ts file level

I could use `Omit<RequestPayload, "controller">`  wiches weird too but maybe more acceptable. Controller is abstracted and taken from controller name and it leads to ts errors without omitting

File: src/controllers/Bulk.ts [hunk 1]
```diff
@@ -62,62 +66,33 @@ export class BulkController extends BaseController {
     bulkData: Array<JSONObject>,
     options: ArgsBulkControllerImport = {}
   ): Promise<{
-    /**
-     * Array of successfully executed actions
-     */
     successes: Array<{
```
Comment (sebtiz13) [inline src/controllers/Bulk.ts:69]: Why remove this doc comments ?

**Review by sebtiz13 — APPROVED** [review]

**Comment by Juiced66** [comment]
For maintainability purpose and after doing pro and cons, I'll stick to maintainability purpose where the query solution is better than little optim nit picking

**Comment by rolljee** [comment]
LIttle UP as this need to be merged 💪🏼

**Comment by Juiced66** [comment]
will close this to open a Doc only PR

**Pull request closed without merging**